Lesson 2

slade1987 opened the pull request.

slade1987 committed:
Calculator

slade1987 committed:
Add O notation

DmitryChitalov commented on Урок 2. Практическое задание/task_1.py:
```diff
@@ -27,3 +27,35 @@
+    if sign == '0':
+        return print("До свидания")
+
+rec(int(input("Введите число")))
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_2.py:
```diff
@@ -17,3 +17,25 @@
+        recur(numb // 10, chet, nechet)
+
+
+recur(int(input("Введите число")), 0, 0)
```
решение можно оптимизировать в неск-ко строк

DmitryChitalov replied:
chet, nechet
что мы знаем о таких именах?

DmitryChitalov commented on Урок 2. Практическое задание/task_3.py:
```diff
@@ -18,3 +18,14 @@
+        return newstr
+
+
+print(recurs(1230, ''))
```
, ''
для чего передавать пустую строку?
нужно обойтись без

DmitryChitalov commented on Урок 2. Практическое задание/task_4.py:
```diff
@@ -10,3 +10,33 @@
+def recurs3(n):
+    return funct(n) if (n == 1) else funct(n) + recurs3(n - 1)
+
+print(recurs3(3))
```
выполнено

DmitryChitalov commented on Урок 2. Практическое задание/task_5.py:
```diff
@@ -19,3 +19,19 @@
+    return getAnsiString(n, n + 9) + "\n" + recurs(n + 10)
+
+
+print(recurs(32))
```
getAnsiString
такие имена - нарушение пеп-8

DmitryChitalov commented on Урок 2. Практическое задание/task_6.py:
```diff
@@ -7,3 +7,25 @@
+
+
+def recurs(num,i=10):
+    ans=int(input("Введите число"))
```
ans=int(input("Введите число"))
все листинги нужно проверить на пеп-8

DmitryChitalov commented on Урок 2. Практическое задание/task_7.py:
```diff
@@ -13,3 +13,17 @@
+    print("Равенство верно")
+else:
+    print("Равенство не верно")
+
```
выполнено

slade1987 committed:
Добавлено пробел
sign =input(..)
sign = input(..)

slade1987 committed:
Реформат кода
Переименовал переменные chet, nechet в even, not_even
-----------------------------------------------

slade1987 committed:
убрал передачу пустой строки в recurs(1230,'')

slade1987 committed:
Убрал лишнюю функцию getAnsiByCode.
Сделал реформат кода переименовал функцию get_ansi_string

slade1987 committed:
Сделал реформат кода согласно pep-8
ans = int(input("Введите число"))
